File: app/src/main/java/com/ocram/qichwadic/framework/net/client/Service.java
package com.ocram.qichwadic.framework.net.client;

import com.ocram.qichwadic.domain.model.Definition;
import com.ocram.qichwadic.domain.model.Dictionary;
import com.ocram.qichwadic.domain.model.SearchResult;

import java.util.List;

import io.reactivex.Flowable;

public interface Service {

    Flowable<List<Dictionary>> getDictionaries();

    Flowable<List<SearchResult>> searchWord(String searchWord, int fromQuechua, String target, int searchType);

    Flowable<List<Definition>> fetchMoreResults(int dictionaryId, String word, int fromQuechua, int searchType, int page);

    Flowable<List<Definition>> getAllDefinitionsByDictionary(int dictionaryId);
}

File: app/src/main/java/com/ocram/qichwadic/core/data/remote/Service.kt
package com.ocram.qichwadic.core.data.remote

import com.ocram.qichwadic.core.data.model.DefinitionEntity
import com.ocram.qichwadic.core.data.model.DictionaryEntity
import com.ocram.qichwadic.core.data.model.SearchResultEntity
import retrofit2.Response

interface Service {

    suspend fun getDictionaries(): Response<List<DictionaryEntity>>

    suspend fun searchWord(searchWord: String, fromQuechua: Int, target: String, searchType: Int): Response<List<SearchResultEntity>>

    suspend fun fetchMoreResults(dictionaryId: Int, word: String, fromQuechua: Int, searchType: Int, page: Int): Response<SearchResultEntity>

    suspend fun getAllDefinitionsByDictionary(dictionaryId: Int): Response<List<DefinitionEntity>>
}
